test/ERC7432.spec.ts

```diff
@@ -201,17 +204,70 @@ describe('Nfts Roles', () => {
         const returnedData = await nftRoles.roleData(role, roleCreator.address, userOne.address, AddressZero, tokenId)
         expect(returnedData).to.equal(customData)
 
-        const returnedExpirationDate = await nftRoles.roleExpirationDate(role, roleCreator.address, userOne.address, AddressZero, tokenId)
+        const returnedExpirationDate = await nftRoles.roleExpirationDate(
+          role,
+          roleCreator.address,
+          userOne.address,
+          AddressZero,
+          tokenId,
+        )
         expect(returnedExpirationDate).to.equal(expirationDate)
       })
-    })
+      it('should decode role custom data with metadata', async () => {
+        const nftMetadata = {
+          name: 'Nft name',
+          description: 'Nft description',
+          image: 'https://example.com/image.png',
+          roles: [
+            {
+              name: 'Role Name',
+              description: 'User of the Nft',
+              inputs: [
+                {
+                  name: 'user',
+                  type: 'address',
+                  description: 'User address',
+                },
+              ],
+            },
+          ],
+        }
+
+        const NftFactory = await ethers.getContractFactory('Nft')
+        const nft = await NftFactory.deploy()
+        await nft.deployed()
 
-    describe('ERC165', async function(){
-      it(`should return true for INftRoles interface id (${NftRolesInterfaceId})`, async function (){
-        expect(await nftRoles.supportsInterface(NftRolesInterfaceId)).to.be.true
+        const customData = defaultAbiCoder.encode(['address'], [userOne.address])
+
+        await expect(
+          nftRoles
+            .connect(roleCreator)
+            .grantRole(role, userOne.address, AddressZero, tokenId, expirationDate, customData),
+        )
+          .to.emit(nftRoles, 'RoleGranted')
+          .withArgs(role, AddressZero, tokenId, userOne.address, expirationDate, customData)
+
+        const returnedData = await nftRoles.roleData(role, roleCreator.address, userOne.address, AddressZero, tokenId)
+
+        const scope = nock('https://example.com').get(`/${tokenId}`).reply(200, nftMetadata)
```

Comment:
Setup the mock server on the `beforeAll` block, so it can be used across all tests, and make it separated from the solidity calls